File: MATTHEW/leetcode/AIcoding/5.Qwen-3/go_solutions/922-sort-array-by-parity-ii.go
package main

func sortArrayByParityII(A []int) []int {
    n := len(A)
    for i, j := 0, 1; i < n && j < n; i += 2 {
        if A[i]%2 == 0 {
            continue
        }
        for j < n && A[j]%2 == 1 {
            j += 2
        }
        if j < n {
            A[i], A[j] = A[j], A[i]
        }
    }
    return A
}

File: leetcode/AIcoding/1.Gemini-2.5/erlang_solutions/922-sort-array-by-parity-ii.erl
-module(solution).
-export([sort_array_by_parity_ii/1]).

sort_array_by_parity_ii(A) ->
    {EvensRev, OddsRev} = lists:foldl(
        fun(X, {Es, Os}) ->
            case X rem 2 of
                0 -> {[X|Es], Os};
                1 -> {Es, [X|Os]}
            end
        end,
        {[], []},
        A
    ),
    Evens = lists:reverse(EvensRev),
    Odds = lists:reverse(OddsRev),
    interleave(Evens, Odds).

interleave([], []) -> [];
interleave([He|Te], [Ho|To]) ->
    [He, Ho | interleave(Te, To)].